src/Http/Routing/test/UnitTests/EndpointRoutingMiddlewareTest.cs

```diff
@@ -393,6 +394,36 @@ public async Task Endpoint_HasBodySizeFeature_SetUsingSizeLimitMetadata(bool isR
         Assert.Equal(expectedRequestSizeLimit, actualRequestSizeLimit);
     }
 
+    [Fact]
+    public async Task Create_WithoutHostBuilder_Success()
+    {
+        // Arrange
+        var httpContext = CreateHttpContext();
+
+        var services = new ServiceCollection();
+        services.AddLogging();
+        services.AddOptions();
+        services.AddSingleton<DiagnosticListener>(s => new DiagnosticListener("Test"));
```

Comment:
UseRouting requires `DiagnosticListener` in DI, but I don't see that in OC source code. I'm confused how it worked before.